### DISCUSSION TITLE
v2.6.0

### DISCUSSION THREAD

**Discussion by mashb1t** [mashb1t/Fooocus#56] [Announcements]
## How to update

This version includes various package updates. If the auto-update doesn't work you can do one of the following:
1. Open a terminal in the Fooocus folder (location of config.txt) and run `git pull`
2. Update packages
   - Windows (installation through zip file): open a terminal in the Fooocus folder (location of config.txt) `..\python_embeded\python.exe -m pip install -r .\requirements_versions.txt` (Windows using embedded python, installation method zip file) or download Fooocus again (zip file attached to this release)
   - other: manually update the packages using `python.exe -m pip install -r requirements_versions.txt` or use the docker image

## What's Changed

* feat: update python dependencies
* i18n: rename tab titles and translations from singular to plural
* i18n: rename document to documentation
* feat: update default models to latest versions
  * animaPencilXL_v400 => animaPencilXL_v500 (https://github.com/lllyasviel/Fooocus/pull/2943)
  * DreamShaperXL_Turbo_dpmppSdeKarras => DreamShaperXL_Turbo_v2_1
  * SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4 => SDXL_FILM_PHOTOGRAPHY_STYLE_V1
*  feat: add preset for pony_v6 (using ponyDiffusionV6XL, discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: add style `Fooocus Pony` (discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: remove `by wlop` from style `Fooocus Masterpiece` as it has been causing unintended watermarks
* feat: add restart sampler ([paper](https://arxiv.org/abs/2306.14878))
* feat: add config option for default_inpaint_engine_version, sets inpaint engine for pony_v6 and playground_v2.5 to None for improved results (incompatible with inpaint engine, discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: add image editor functionality to mask upload (same as for inpaint, now correctly resizes and allows more detailed mask creation, discussion in https://github.com/mashb1t/Fooocus/discussions/44)
* feat: add persistent model cache for metadata. Use `--rebuild-hash-cache` to manually rebuild the cache for all non-cached hashes (optional, cache will otherwise be lazy-loaded. Remove `--rebuild-hash-cache` after executing once)
* refactor: rename `--enable-describe-uov-image` to `--enable-auto-describe-image` to better reflect its purpose (now also works for enhance image upload)
* build(deps): bump docker/build-push-action from 5 to 6 by @dependabot in https://github.com/lllyasviel/Fooocus/pull/3223

**Full Changelog**: https://github.com/mashb1t/Fooocus/compare/v2.5.2...v2.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mashb1t/Fooocus/releases/tag/v2.6.0'>v2.6.0</a>.</em>

**Comment by supermimai** [comment]
Hello
Trying to update with your suggestion , the "git pull", etc, I was not sure about the results, so I take the win64_2-6-0 but I have this error ( last line ) : 
E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Fast-forward merge
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.6.0 (mashb1t)
[Cleanup] Attempting to delete content of temp dir C:\Users\fabri\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/mashb1t/misc/resolve/main/xl-to-v1_interposer-v4.0.safetensors" to E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\models\vae_approx\xl-to-v1_interposer-v4.0.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 5.40M/5.40M [00:00<00:00, 50.8MB/s]
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_v8Rundiffusion.safetensors" to E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors

100%|██████████████████████████████████████████████████████████████████████████████| 6.62G/6.62G [01:03<00:00, 113MB/s]
Total VRAM 24576 MB, total RAM 65417 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : native
VAE dtype: torch.bfloat16
Traceback (most recent call last):
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\launch.py", line 147, in <module>
    from webui import *
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\webui.py", line 10, in <module>
    import modules.async_worker as worker
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\modules\async_worker.py", line 3, in <module>
    from extras.inpaint_mask import generate_mask_from_image, SAMOptions
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\extras\inpaint_mask.py", line 6, in <module>
    from extras.GroundingDINO.util.inference import default_groundingdino
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\Fooocus\extras\GroundingDINO\util\inference.py", line 9, in <module>
    import supervision as sv
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\supervision\__init__.py", line 7, in <module>
    from supervision.detection.tools.polygon_zone import PolygonZone, PolygonZoneAnnotator
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\supervision\detection\tools\polygon_zone.py", line 10, in <module>
    from supervision.draw.utils import draw_polygon, draw_text
  File "E:\fooocus\Fooocus_mashb1t\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\supervision\draw\utils.py", line 111, in <module>
    text_font: int = cv2.FONT_HERSHEY_SIMPLEX,
AttributeError: module 'cv2' has no attribute 'FONT_HERSHEY_SIMPLEX'

**Reply by mashb1t** [reply]
@supermimai let me check, i can reproduce the issue

**Reply by mashb1t** [reply]
@supermimai fixed, sorry for the inconvenience. cv2 did not correctly get downloaded when creating the zip file.
The file has been updated in-place and you can now download the corrected version at https://github.com/mashb1t/Fooocus/releases/download/v2.6.0/Fooocus_win64_2-6-0.7z

**Reply by supermimai** [reply]
@mashb1t  works now better, when I run it from the zip, but when I override the zip in my past installation I have plenty of errors than I ( unfornutly ) don't understand how to fix them. 
I will restore and change again my config from the new zip 2.6.0 I think I will found how to do it ! :) thanks !!!

**Comment by Turpo69** [comment]
Bug? Input image tab Metadata is not reading metadata from images at all.

**Reply by Turpo69** [reply]
![Clip_6](https://github.com/user-attachments/assets/f543c2f6-24af-4c4b-bb31-746948cceb04)

**Reply by mashb1t** [reply]
Works for me without issues. Did you generate this image with Fooocus and did you turn on metadata in the config or the advanced tab > developer debug mode? Any terminal outputs?

**Reply by Turpo69** [reply]
it's generated with this version. All I know is that importing metadata worked just fine in rc2. I also tried images done with earlier versions.

**Reply by mashb1t** [reply]
@Turpo69 can't help without you providing terminal logs

**Reply by Turpo69** [reply]
I don't even know what terminal logs you mean? Maybe unzipping to overwrite old files didn't work 100% this time. I think I have to unzip to clean folder to see if it works that way.

**Reply by Turpo69** [reply]
Clean folder install didn't help. I suppose that I can't be only one with this problem then.

**Reply by mashb1t** [reply]
Wherever you start Fooocus there have to be logs in the console / terminal, which can provide clarity. Without them you can not be helped as you may have modified your files. Both Colab and local versions from a clean install do not have this issue. Please double-check logs and open a new issue if necessary.

**Reply by Turpo69** [reply]
Yep, console just says: Could not find metadata in the image! I just find out that it's jpg problem. It reads metadata from png's just fine. I use jpeg as default. Only when I generate images to post in civitai I use png and a1111 metadata scheme.

**Reply by mashb1t** [reply]
@Turpo69 i can reproduce the issue in Colab, let me check locally

**Reply by mashb1t** [reply]
fixed & merged in both this fork and upstream, thank you for reporting!

**Reply by Turpo69** [reply]
Thank you for all the hard work you put into Fooocus.

**Comment by coarturo280385** [comment]
Hola!! Mashb1t, en verdad necesito de tu apoyo. Tengo 4 días intentando ejecutar en programa y no me descarga o ejecuta para iniciar el proceso de instalación...
De mi parte tengo control total de Windows 11, pensaba que algo en mi sistema estuviera bloqueando la herramienta Foooocus pero no, tambien deshavilited el anti Virus, tambien desistale mi Python, nada a funcionado...... Adicional actualizo todas las librerias y por nda se ejecuta el programa.
Tambien instale la version de Fooocus en la plataforma PINOKIO, ejecuta toda la instalación de los aplicativos pero tampoco se despliega la ventana del entorno de trabajo.... De mi parte instale todas las versiones de Fooocus pero con todas optengo el mismo error, ya sea como administrador o no.
Deve de sir una falla en mi computadora pero no la encuentro ya que nadie le ha pasado este tipo de error.
Mi computadora es gama alta.

Comparto el ERROR:

D:\Fooocus_win64_2-6-0>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.6.5 (mashb1t)
Traceback (most recent call last):
  File "D:\Fooocus_win64_2-6-0\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "D:\Fooocus_win64_2-6-0\Fooocus\launch.py", line 88, in <module>
    from modules.hash_cache import init_cache
  File "D:\Fooocus_win64_2-6-0\Fooocus\modules\hash_cache.py", line 7, in <module>
    from modules.util import sha256, HASH_SHA256_LENGTH, get_file_from_folder_list
  File "D:\Fooocus_win64_2-6-0\Fooocus\modules\util.py", line 8, in <module>
    import cv2
  File "D:\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\cv2\__init__.py", line 181, in <module>
    bootstrap()
  File "D:\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\cv2\__init__.py", line 153, in bootstrap
    native_module = importlib.import_module("cv2")
  File "importlib\__init__.py", line 126, in import_module
    if sys.path[0] == BASE_DIR or os.path.realpath(sys.path[0]) == BASE_DIR:
ImportError: DLL load failed while importing cv2: No se puede encontrar el módulo especificado.

D:\Fooocus_win64_2-6-0>pause
Presione una tecla para continuar . . .

**Reply by mashb1t** [reply]
@coarturo280385 see https://github.com/opencv/opencv-python/issues/856#issuecomment-1713214860 and connected issues, install Windows Media Pack.

**Reply by coarturo280385** [reply]
¡¡¡Muchas Gracias!! despues de 4 dias de intensos analisis de fallas, la solución es instale Windows Media Pack para que el programa se ejecute.
Adicional, estoy configurando tu versión mejorada de Fooocus_win64_2-6-0.... Tienes a la mano unos tutoriales al respecto..  

Estoy muy  agradecido cordial saludos.

**Comment by ViktorVorobyev** [comment]
I tried to download the pony preset, it gives an error:

Loaded preset: C:\AI\Fooocus_win64_2-6-0\Fooocus\presets\pony_v6.json
Downloading: "https://huggingface.co/mashb1t/fav_models/resolve/main/fav/ponyDiffusionV6XL.safetensors" to C:\AI\Fooocus_win64_2-6-0\Fooocus\models\checkpoints\ponyDiffusionV6XL.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 6.46G/6.46G [10:09<00:00, 11.4MB/s]
Traceback (most recent call last):
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\AI\Fooocus_win64_2-6-0\python_embeded\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "C:\AI\Fooocus_win64_2-6-0\Fooocus\webui.py", line 935, in preset_selection_change
    preset_prepared['base_model'], preset_prepared['checkpoint_downloads'] = launch.download_models(
  File "C:\AI\Fooocus_win64_2-6-0\Fooocus\launch.py", line 140, in download_models
    load_file_from_url(url=url, model_dir=config.path_vae, file_name=file_name)
  File "C:\AI\Fooocus_win64_2-6-0\Fooocus\modules\model_loader.py", line 19, in load_file_from_url
    os.makedirs(model_dir, exist_ok=True)
  File "os.py", line 210, in makedirs
  File "ntpath.py", line 211, in split
TypeError: expected str, bytes or os.PathLike object, not list
